src/components/TodoList/TodoItem.jsx

```diff
@@ -28,60 +29,33 @@ const TodoItem = (todoItem) => {
             .catch((err) => console.log(err));
     };
 
-    const changeTodoColor = (priority) => {
-        switch (priority) {
-            case "high":
-                setGradientColor("from-red-400");
-                setBorderColor("border-red-500");
-                break;
-            case "medium high":
-                setGradientColor("from-amber-400");
-                setBorderColor("border-amber-500");
-                break;
-            case "medium":
-                setGradientColor("from-yellow-300");
-                setBorderColor("border-yellow-400");
-                break;
-            case "medium low":
-                setGradientColor("from-green-300");
-                setBorderColor("border-green-400");
-                break;
-            case "low":
-                setGradientColor("from-green-500");
-                setBorderColor("border-green-600");
-                break;
-        }
-        if (todoItemData.completed === true) {
-            setGradientColor("from-gray-500");
-            setBorderColor("border-black-500");
-        }
-    };
-
-    useEffect(() => {
-        changeTodoColor(todoItemData.priority);
-    }, [todoItemData]);
-
     return (
         <div
-            className={`flex justify-between space-x-5 border-solid border-2 ${borderColor} rounded-lg my-2 h-14 items-center p-4 bg-gradient-to-l ${gradientColor}`}>
+            className={`flex justify-between space-x-5 border-solid border-2 ${todoContainerColor[priority].border} rounded-lg my-2 h-14 items-center p-4 bg-gradient-to-l ${todoContainerColor[priority].gradient}`}>
             <PriorityIcon todoItemData={todoItemData} />
-            <p className="table-cell font-medium">{todoItemData.title}</p>
-            <p className="table-cell italic">{todoItemData.description}</p>
-            <button
-                className="bg-red-500 text-white font-medium px-2 py-1 rounded hover:bg-red-900 h-8"
-                onClick={() => handleDeleteTodo(todoItemData)}>
-                <AiOutlineDelete />
-            </button>
+            <p className="todo-text font-medium">{todoItemData.title || emptyField}</p>
+            <p className="todo-text italic">{todoItemData.description}</p>
             <button
-                className="bg-blue-500 text-white font-medium px-2 py-1 rounded hover:bg-blue-900 h-8"
+                title="Edit Todo"
+                className={`bg-${
+                    todoItemData.completed ? "blue-500 opacity-50" : "blue-500"
+                } text-white font-medium px-2 py-1 rounded hover:bg-blue-900" h-8`}
+                disabled={todoItemData.completed}
                 onClick={() => handleEditTodo(todoItemData)}>
                 <AiOutlineEdit />
             </button>
             <button
+                title="Complete Todo"
                 className="bg-blue-500 text-white font-medium px-2 py-1 rounded hover:bg-blue-900 h-8"
                 onClick={() => toggleCompleteTodo()}>
```

Comment:
This function is not taking any parameters. You can just pass it.
```
onClick={toggleCompleteTodo}
```
otherwise every render a new anonymous function is being defined, which takes more memory. Not really a performance benefit here as the app is not big enough to notice an issue but it is a good practice to avoid.

Same goes for all the other function in this component.
You have the todoItem data from the prop. You can define all the functions outside the return and pass the functions themselves.

```
// Closure
function Component({ item }) {
  const onDelete = () => axios.delete(`/todo/${item?.id}`)

  return (
     <button onClick={onDelete}>DELETE</button>
  )
}
```
Here is a video explaining it + an interview question that we use here in Dreamix: https://www.youtube.com/watch?v=vKJpN5FAeF4